File: 프로그래머스/3/64062. 징검다리 건너기/징검다리 건너기.py
def solution(stones, k):
    answer = 0
    start = 1
    end = 200000000
    
    while(start<=end):
        mid = (start + end)//2
        flag = 0
        cnt = 0
        for stone in stones:
            if stone-mid <=0:
                cnt +=1
                if cnt >= k:
                    flag = 1
                    break
            else:
                cnt = 0
        if flag:
            end = mid-1
        else:
            start = mid+1
    
    return start

File: 프로그래머스/3/64062. 징검다리 건너기/징검다리 건너기.js
function solution(stones, k) {
    let start = 1
    let end = 200000000

    while(start<=end){
        let mid = Math.ceil((start + end)/2)
        let cnt = 0
        for(let stone of stones){
            if (stone-mid <= 0){ 
                cnt++
            }else cnt = 0
            
            if(cnt >= k){
                end = mid-1
                break
            }
        } 
        if(cnt < k){
            start = mid+1
        }
    }
    return start;
}